interpreter/functions/formatting.go

```diff
@@ -0,0 +1,164 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package functions
+
+import (
+	"errors"
+	"fmt"
+	"strconv"
+	"strings"
+	"unicode"
+
+	"github.com/google/cel-go/common/types/ref"
+)
+
+// FormatCallback is an interface that allows user-defined behavior
+// for formatting clause implementations, as well as argument retrieval.
+type FormatCallback interface {
+	String(ref.Val, string) (string, error)
```

Comment:
Explain somewhere here what these functions return? From this interface declaration it is not obvious to me. If all the functions have uniform semantics, I'd be fine with it being stated on the type declaration.

Reply:
Right, I've added docs.